test/ca/mcgill/jetuml/layouttests/TestLayoutObjectDiagram.java

```diff
@@ -1,17 +1,197 @@
+import org.junit.jupiter.params.provider.CsvSource;
+import org.junit.jupiter.params.provider.ValueSource;
+
+import ca.mcgill.cs.jetuml.diagram.Edge;
+import ca.mcgill.cs.jetuml.diagram.Node;
+import ca.mcgill.cs.jetuml.diagram.edges.NoteEdge;
+import ca.mcgill.cs.jetuml.diagram.edges.ObjectReferenceEdge;
+import ca.mcgill.cs.jetuml.geom.Rectangle;
+import ca.mcgill.cs.jetuml.viewers.edges.EdgeViewerRegistry;
+import ca.mcgill.cs.jetuml.viewers.nodes.NodeViewerRegistry;
+import ca.mcgill.cs.jetuml.viewers.nodes.ObjectNodeViewer;
+
+/**
  * This class tests that the layout of a manually-created diagram file corresponds to expectations.
  */
 public class TestLayoutObjectDiagram extends AbstractTestDiagramLayout
 {
 	private static final Path PATH = Path.of("testdata", "testPersistenceService.object.jet");
 
+	/**
+	 * We add two pixels to the length of an edge to account for the stroke width and/or the arrow head.
+	 */
+	private static final int BUFFER = 2; 
+	
 	TestLayoutObjectDiagram() throws IOException
 	{
 		super(PATH);
 	}
+	
+	/**
+	 * Tests that nodes are in the position that corresponds to their position value 
+	 * in the file. 
+	 */
+	@ParameterizedTest
+	@CsvSource({":Type1, 240, 130",
+				"object2:, 540, 150",
+				":Type3, 610, 300",
+				"A note, 280, 330"})
+	void testNamedNodePosition(String pNodeName, int pExpectedX, int pExpectedY)
+	{
+		verifyPosition(nodeByName(pNodeName), pExpectedX, pExpectedY);
+	}
+	
+	/**
+	 * Tests that the unnamed node is in the position that corresponds to its position value in the file. 
+	 */
+	@Test
+	void testUnnamedNodePosition()
+	{
+		verifyPosition(nodeByName(""), 440, 290);
+	}
+	
+	/**
+	 * Tests that all object nodes that are supposed to have the default dimension
+	 * actually do. 
+	 */
+	@ParameterizedTest
+	@ValueSource(strings = {"object2:", ":Type3"})
+	void testObjectNodesDefaultDimension(String pNodeName)
+	{
+		verifyObjectNodeDefaultDimensions(nodeByName(pNodeName));
+	}
+	
+	/**
+	 * Tests that the object nodes that are supposed to be expanded, actually are. 
+	 */
+	@ParameterizedTest
+	@ValueSource(strings = {":Type1", ""})
+	void testObjectNodeExpandedVertically(String pNodeName)
+	{
+		try
+		{
+			final int DEFAULT_HEIGHT = getStaticIntFieldValue(ObjectNodeViewer.class, "DEFAULT_HEIGHT");
+			Rectangle bounds = NodeViewerRegistry.getBounds(nodeByName(pNodeName));
+			assertTrue(bounds.getHeight() > DEFAULT_HEIGHT);
+		} 
+		catch(ReflectiveOperationException e)
+		{
+			fail();
+		}
+	}
+	
+	/**
+	 * Tests that the collaboration edge connects to its node boundaries. 
+	 */
+	@Test
+	void testCollaborationEdge()
+	{
+		Rectangle boundsUnnamedNode = NodeViewerRegistry.getBounds(nodeByName(""));
+		Rectangle boundsNodeObject2 = NodeViewerRegistry.getBounds(nodeByName("object2:"));
+		Rectangle edgeBounds = EdgeViewerRegistry.getBounds(edgeByMiddleLabel("e1"));
+		assertWithTolerance(boundsNodeObject2.getMaxY(), BUFFER, edgeBounds.getY());

```

Comment:
Why don't we create a convenience method `assertWithDefaultTolerance` which encapsulates the buffer value, and use this everywhere instead. Right now I can't think of other tolerance values we might need.